posthog/api/decide.py

```diff
@@ -122,6 +123,7 @@ def get_decide(request: HttpRequest):
             team = user.teams.get(id=project_id)
         if team:
             response["featureFlags"] = get_active_feature_flags(team, data["distinct_id"])
+            response["featureFlagDetails"] = get_active_feature_flag_details(team, data["distinct_id"])
```

Comment:
I think it's going to be confusing for a consumer of this API to get `featureFlags` and this new `featureFlagDetails` key in the response – I don't think it'll be obvious which one to use.

Instead of adding a new key for this, can we instead add additional information to the get_active_feature_flags call to capture this data?

Reply:
I agree it'd be cleaner if all the info was under one endpoint. With changes to both frontend and backend, I could get it to work on PostHog after some extensive testing.

- We can change how we consume our own API, and with our release cycle of rolling out frontend and backend at the same time they'll always be in sync. Not always the case for user consumers (that is, people using feature flags in their apps).
- For user consumers, we can update how the method `isFeatureEnabled(feature: string)` works in our JS library - see https://github.com/PostHog/posthog-js/blob/master/src/posthog-featureflags.js#L57
  - Luckily, our Python API accesses full feature flag info by using the user's secret **personal API key**, which returns more than just key names, so fewer/no changes will be needed there.

Our official recommendation for users that are _not_ using JS or Python helpers is to hit `POST /decide`. Currently, the response looks like:

```json
{
  "config": {
    "enable_collect_everything": true
  },
  "editorParams": {},
  "isAuthenticated": false,
  "supportedCompression": [
    "gzip",
    "lz64"
  ],
  "featureFlags": [
    "my-awesome-flag-1",
    "my-awesome-flag-2"
  ]
}
```

The question then becomes — what are users currently doing with the API response? If `featureFlags` changes from an array of strings to an object, or an array of objects, does that break for them? 

**Option 1: Single keyed object**

This would break accessing via array methods.

```jsonc
{
  // ...
  "featureFlags": {
    "my-awesome-normal-flag-1": true,
    "my-awesome-normal-flag-2": true,
    "my-variant-flag": "cool-variant",
  }
}
```

**Option 2: Array of objects**
```jsonc
{
  // ...
  "featureFlags": [
    {
      "key": "my-awesome-normal-flag-1",
      "value": true
    },
    {
      "key": "my-awesome-normal-flag-2",
      "value": true
    },
    {
      "key": "my-variant-flag",
      "value": "cool-variant"
    }
  ]
}
```

A possible solution would be to mix data types in the `decide` response. The advantage of this is that if a user tries to access via array methods, like `featureFlags.includes("string")`, it would still work for a normal flag. It'd then look like:

**Option 3: Mixed data types**
```jsonc
{
  //...
  "featureFlags": [
    "my-awesome-normal-flag-1",
    "my-awesome-normal-flag-2",
    {
      "key": "my-variant-flag",
      "value": "cool-variant"
    }
  ]
}
```

... which seems a little weird, but maybe OK?

Reply:
I think this format in `/decide` makes the most sense:

```
{
  // ...
  "featureFlags": {
    "my-awesome-normal-flag-1": true,
    "my-awesome-normal-flag-2": true,
    "my-variant-flag": "cool-variant",
    "my-awesome-normal-flag-2": { "someother": "json property data that can be configured" } ,
  }
}
```

However we're forever going to have the case of old clients that won't expect this. There's a solution to that: versioning. 

We could have newer versions of posthog-js call `/decide?v=2` and that would return the flags as described above. We would then make sure that the API exposed by `posthog-js` (`onFeatureFlags(flags: string[])`, etc) would not change in a breaking way.